rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestBookKeeper.java

```diff
@@ -138,4 +142,52 @@ public void testGetPrestoClusterManagerInValidInstance() throws Exception
 
     ClusterManager manager = bookKeeper.getClusterManagerInstance(type, conf);
   }
+
+  @Test
+  public void testGetFileInfoWithInvalidationEnabled() throws Exception
+  {
+    Path backendFilePath = new Path(BookKeeperTestUtils.getDefaultTestDirectoryPath(conf), BACKEND_FILE_NAME);
+    DataGen.populateFile(backendFilePath.toString());
+    int expectedFileSize = DataGen.generateContent(1).length();
+
+    CacheConfig.setFileInvalidationEnabled(conf, true);
+
+    BookKeeper bookKeeper = new CoordinatorBookKeeper(conf, metrics);
+    FileInfo info = bookKeeper.getFileInfo(backendFilePath.toString());
+
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+
+    //Rewrite the file with half the data
+    DataGen.populateFile(backendFilePath.toString(), 2);
+
+    expectedFileSize = DataGen.generateContent(2).length();
+
+    info = bookKeeper.getFileInfo(backendFilePath.toString());
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+  }
+
+  @Test
+  public void testGetFileInfoWithInvalidationDisabled() throws Exception
+  {
+    Path backendFilePath = new Path(BookKeeperTestUtils.getDefaultTestDirectoryPath(conf), BACKEND_FILE_NAME);
+    DataGen.populateFile(backendFilePath.toString());
+    int expectedFileSize = DataGen.generateContent(1).length();
+
+    CacheConfig.setFileInvalidationEnabled(conf, false);
+
+    BookKeeper bookKeeper = new CoordinatorBookKeeper(conf, metrics);
+    FileInfo info = bookKeeper.getFileInfo(backendFilePath.toString());
+
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
+        " Got FileSize: " + info.getFileSize() + " Expected Value : " + expectedFileSize);
+
+    //Rewrite the file with half the data
+    DataGen.populateFile(backendFilePath.toString(), 2);
+
+    info = bookKeeper.getFileInfo(backendFilePath.toString());
+    assertTrue(info.getFileSize() == expectedFileSize, "FileSize was not equal to the expected value." +
```

Comment:
Is this proving that the file size is still 1 in the metadata ?

Reply:
This is to show that although we have changed the file in the backend (which is half the original size), the fileinfo still has the old value. (Data is being cached)